lib/chef/win32/handle.rb

```diff
@@ -21,27 +21,37 @@
 require_relative "api/system"
 require_relative "error"
 
+require "chef-powershell"
+
 class Chef
   module ReservedNames::Win32
     class Handle
       extend Chef::ReservedNames::Win32::API::Process
 
+      include ChefPowerShell::ChefPowerShellModule::PowerShellExec
+
       # See http://msdn.microsoft.com/en-us/library/windows/desktop/ms683179(v=vs.85).aspx
       # The handle value returned by the GetCurrentProcess function is the pseudo handle (HANDLE)-1 (which is 0xFFFFFFFF)
       CURRENT_PROCESS_HANDLE = 4294967295
 
       def initialize(handle)
         @handle = handle
+        @is_2012 ||= get_proc_os
         ObjectSpace.define_finalizer(self, Handle.close_handle_finalizer(handle))
       end
 
       attr_reader :handle
 
+      def get_proc_os
```

Comment:
this doesn't get the os. Lets call it like `win2002notr2?` if we want it to do this.

Reply:
done